src/query/service/src/pipelines/processors/transforms/hash_join/probe_join/cross_join.rs

```diff
@@ -35,8 +35,19 @@ impl HashJoinProbeState {
         if build_num_rows == 0 || input_num_rows == 0 {
             return Ok(vec![]);
         }
-        let probe_block = input.project(&self.probe_projections);
+        let mut probe_block = input.project(&self.probe_projections);
         let build_block = DataBlock::concat(build_blocks)?;
+        if build_num_rows == 1 {
+            for col in build_block.columns() {
```

Comment:
can use `build_block.slice(0,1)`

Reply:
or just return the `build_block `

Reply:
Need to construct `Value::Scalar` instead of `Value::Column`.